test/text/simple_text_displayer_unit.js

```diff
@@ -127,6 +127,33 @@ describe('SimpleTextDisplayer', () => {
           [shakaCue]);
     });
 
+    it('flattens nested cue payloads correctly', () => {
```

Comment:
This test looks great.  So I'm not concerned about the validity of the code change, so much as the clarity and readability.

Reply:
Thanks, after the updates based on your other comments I think it should be easier to follow.